### DISCUSSION TITLE
Should I add App_Data to source control ?

### DISCUSSION THREAD

**Discussion by hung-doan** [OrchardCMS/OrchardCore#10371] [Q&A]
Currently I'm  using : 
- Local file system shell (https://docs.orchardcore.net/en/dev/docs/reference/core/Shells/)
- Local file store for Content Definition (https://docs.orchardcore.net/en/dev/docs/guides/content-definitions/)

It seems these settings are all stored in App_Data .
So that my questions is should I add App_Data to my git source control ?.
If yes, which file should I added to source control ?. 
e.g: 
```
App_Data\tenants.json
App_Data\Sites\Default\ContentDefinition.json
...
```

**Comment by Piedone** [comment]
While you can do that fairly OK if you're a sole developer if the project gets more complex and especially if more than one person works on it this will become an issue. Foremost, the DB being a binary file (if you're using SQLite but otherwise this won't work at all) makes it impossible to properly diff changes, so two people won't be able to change it at the same time. This would quickly get messy anyway once you want to deploy the site to a prod environment and the DB of the prod site and your local copy will diverge. In short, don't do it :).

Rather, you could store all the data and configuration (including content type definitions, though for custom ones migration classes are better I think) in [recipes](https://docs.orchardcore.net/en/latest/docs/reference/modules/Recipes/). You'd use these during local development, potentially together with [Auto Setup](https://docs.orchardcore.net/en/dev/docs/reference/modules/AutoSetup/). Then, you could use a DB only in prod.

**Comment by hishamco** [comment]
> So that my questions is should I add App_Data to my git source control ?

No need
